src/kava.js

```diff
@@ -520,10 +520,13 @@ class Kava extends BasePlugin {
       this._fragLoadedFiredOnce = true;
     }
     this._updateFragLoadedStats(event);
-    this._handleFragPerformanceObserver(event.payload.url);
+    const succHandle = this._handleFragPerformanceObserver(event.payload.url);
+    if (!succHandle && this._performanceObserver) {
+      this._pendingFragLoadedEvents.push(event.payload.url);
+    }
   }
 
-  _handleFragPerformanceObserver(url: string): void {
+  _handleFragPerformanceObserver(url: string): boolean {
```

Comment:
the bug is that you are passing different things to `_handleFragPerformanceObserver` in `_onFragLoaded` and on `_handleNewPerformanceEntries`, in the first it's URL and in the later it's performance entry object.
The issue is that you have a never ending while condition that never get's resolved https://github.com/kaltura/playkit-js-kava/blob/4337ef2d8433d07427578d746ac5f03b29b9054c/src/kava.js#L98-L100
Or at least potentially it never get resolved as you had it now.
I think this is not safe.

Reply:
Because you moved the condition now that caused the issue: 
![image](https://user-images.githubusercontent.com/5461862/61589869-fe486400-abb8-11e9-997a-8032ebfeb528.png)
so it won't happen now, but I'm wondering how this passed flow checks.
Also, the definition is incorrect cause you want a URL array, but you define it like this: https://github.com/kaltura/playkit-js-kava/blob/4337ef2d8433d07427578d746ac5f03b29b9054c/src/kava.js#L38